### ISSUE TITLE
error: the option `Z` is only accepted on the nightly compiler

### ISSUE THREAD

**Issue by Enselic** [cargo-public-api/cargo-public-api#73]
This night [CI](https://github.com/Enselic/cargo-public-api/actions/runs/2654032839) failed. Sure enough, I can reproduce it locally by doing:

```
% cargo run -- --manifest-path cargo-public-api/Cargo.toml
Error: Failed to build rustdoc JSON. Stderr:     Checking public-api v0.12.2 (/Users/martin/src/cargo-public-api/public-api)
    Checking rustdoc-json v0.2.1 (/Users/martin/src/cargo-public-api/rustdoc-json)
 Documenting cargo-public-api v0.12.2 (/Users/martin/src/cargo-public-api/cargo-public-api)
error: the option `Z` is only accepted on the nightly compiler

error: could not document `cargo-public-api`

Caused by:
  process didn't exit successfully: `/Users/martin/.rustup/toolchains/stable-aarch64-apple-darwin/bin/rustdoc --edition=2021 --crate-type lib --crate-name cargo_public_api cargo-public-api/src/lib.rs -o /Users/martin/src/cargo-public-api/target/doc --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat -Z unstable-options --output-format json --cap-lints warn -C metadata=a8187698374ba899 -L dependency=/Users/martin/src/cargo-public-api/target/debug/deps --extern ansi_term=/Users/martin/src/cargo-public-api/target/debug/deps/libansi_term-2dcfebadd4a11f02.rmeta --extern anyhow=/Users/martin/src/cargo-public-api/target/debug/deps/libanyhow-ff31ead167d75e4a.rmeta --extern atty=/Users/martin/src/cargo-public-api/target/debug/deps/libatty-03508689ddba4741.rmeta --extern clap=/Users/martin/src/cargo-public-api/target/debug/deps/libclap-85bbfd8e37db05d7.rmeta --extern diff=/Users/martin/src/cargo-public-api/target/debug/deps/libdiff-2132f39c67c267ab.rmeta --extern public_api=/Users/martin/src/cargo-public-api/target/debug/deps/libpublic_api-ea8caa28a08ae247.rmeta --extern rustdoc_json=/Users/martin/src/cargo-public-api/target/debug/deps/librustdoc_json-149a0e783af514d2.rmeta --crate-version 0.12.2` (exit status: 1)
```

For some reason, the stable rustdoc (`/Users/martin/.rustup/toolchains/stable-aarch64-apple-darwin/bin/rustdoc`) is used even though we pass `+nightly` to `cargo`.

Maybe related to the recently released `rustup 1.25.0`?

**Comment by Enselic** [comment]
Possibly related: https://github.com/rust-lang/rustup/issues/3029

**Comment by Enselic** [comment]
Closed by #74
